content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,504 @@
+########################
+
+:lang: en
+:date: 2017-04-13
+:url: getting-started-with-elm
+:summary: What is Elm? how does it work? How can I try it really fast?
+:status: draft
+
+Introduction
+============
+
+The past two weeks we have been traveling and meeting a bunch of
+developers.
+
+It was a chance for us to tell them about Elm_ through talks at
+conferences and to present a short one-hour workshop.
+
+You can find the slides of the two talks here:
+
+- `a short introduction to Elm by Niko <http://slides.com/n1k0/elm>`_
+- `Talking JSON with Elm by Natim <http://natim.github.io/django-slides/elm-json/>`_
+
+However, this article is a summary of the take aways of this workshop
+for those who would like to know a bit more about Elm.
+
+
+What is Elm and why should I be interested?
+===========================================
+
+Elm_ is a development platform, providing a programming language, a
+compiler, an architecture and tooling around. It is focused on making
+sure your app state and the HTML that reflects it are always in sync.
+
+If you have been doing any frontend development during the last couple
+of years, you might have seen that the JavaScript ecosystem is a bit
+wild. Keeping up-to-date with the growing number of competing
+frameworks can take a lot of energy, not to mention the challenge in
+making sure you app works on all the available browser
+versions.
+
+.. _React: https://facebook.github.io/react/
+.. _Angular: https://angularjs.org/
+.. _`Vue.js`: https://vuejs.org/
+.. _Ember: https://www.emberjs.com/
+
+In addition, writing tests for the front-end can be tedious and
+supporting Flow_ or TypeScript_, while tremendously useful, is yet
+another layer to setup, configure and maintain in your stack.
+
+.. _Flow: https://flow.org/en/
+.. _TypeScript: https://www.typescriptlang.org/
+
+Elm brings functional programming to your browser, the language is
+statically typed so with Elm you won't see any run-time errors caused
+by wrong type. Being purely functional, Elm let you write code that is
+completely decoupled, it makes your code much more reusable and more
+easy to refactor.
+
+
+How is Elm different?
+=====================
+
+Most of the time, using Elm_, the two lines it takes to load your app
+is all the JavaScript you will write. Your application will be written
+in Elm_ and compiled into JavaScript.
+
+All the mistakes you will do will be caught at compilation time, which
+means that you will barely never encounter uncaught exceptions at
+runtime in the browser.
+
+Because you will have a compilation step, Elm_ will force you to handle
+every possible case of your model state. It will make sure you covered
+them on the rendering side.
+
+The tooling is great. elm-live_ will automatically update the browser
+code to reflect your code update in realtime. It comes with a debugger
+that will show you the list of events and state of your application at
+a given time, you can also go back in time and replay previous
```

Comment:
I think this is a [comma splice](https://en.wikipedia.org/wiki/Comma_splice). (Which I just learned are acceptable in French, according to this Wikipedia article!) Considering that the second clause is a refinement of the first clause, I'd probably change the comma into an em-dash and change "you can also" to "you can even".